tests/integration/io-threads.tcl

```diff
@@ -0,0 +1,29 @@
+tags {"io-threads external:skip"} {
+    test "Disable io-threads" {
+        start_server {overrides {"io-threads" {1} "io-threads-do-reads" {no}}} {
+            set master_pid [srv 0 pid]
+            set pstree_out [exec ps -T -p $master_pid]
+            assert_no_match "*io_thd_*" $pstree_out
+            assert_match "*io_threads_maximum_num:1*" [r info server]
+            assert_match "*io_threads_active_num:1*" [r info server]
+
+            assert_equal {OK} [r set k v]
+            assert_equal {1} [r DBSIZE]
+        }
+    }
+
+    test "Enable io-threads" {
```

Comment:
Can we add a test case that validates 

1. the active thread count can increase to the max
2. the active thread count can decay back to 1

Writing a reliable test for this feature is tricky and we might need to introduce additional `DEBUG` hooks but I think it will be worth the effort.

Reply:
> Writing a reliable test for this feature is tricky and we might need to introduce additional DEBUG hooks but I think it will be worth the effort

Can you provide more guidance of how to introduce additional DEBUG hooks? In my local, I used `memtier_benchmark -c 25 -t 4`  to simulate the multiple clients/requests, I am not sure if the TCL test framework had the similar API or where can I test the correctness, debug print a message of the `io_threads_active_num`?

Reply:
Kindly ping @PingXie

Reply:
I was thinking about adding some new `DBBUG` command to modify the engine behavior (it looks like you made a change in debug.c but it doesn't look like you've completed it). After seeing #344, I think a real unit test would be more preferable.

Reply:
@PingXie Thanks for your effort to review this. 

> I was thinking about adding some new DBBUG command to modify the engine behavior (it looks like you made a change in debug.c but it doesn't look like you've completed it)

I just add a monitor for `DEBUG io-threads` command, sorry, I didn't get your point to modify the engine behavior, can you be more specific? 

> After seeing https://github.com/valkey-io/valkey/pull/344, I think a real unit test would be more preferable.

Do you want to add the unit test in this PR, or we can add the unit test when new test framework merged?

Reply:
>can you be more specific?

No worries. I was thinking of manipulating the data structure like `clients_pending_write` in a new `DEBUG` command to simulate different workload. Now that we will have a true unit test framework, I don't think we should bother with this hack at all. Do you mind reverting the changes in debug.c? 

>Do you want to add the unit test in this PR, or we can add the unit test when new test framework merged?

I think @madolson is actively working on #344 and it is also close to completion. That said, I am fine with adding the tests in a separate PR and if you decide so, could you open an issue and assign it to yourself?

Reply:
I like the idea of adding some DEBUG commands so we can properly do TCL tests of IO threading features. It probably doesn't make sense to play with the pending writes directly, but something like `DEBUG set-active-io-thread-count` could be a good DEBUG command, even outside of just TCL testing, so that we can force IO threads on and off. E.g. I remember debugging some TLS related issues with IO threading that only happened when IO threads turned off, and it required me to run and stop valkey-benchmark repeatedly.

So I would suggest:

1. Unit testing the "pending writes" to "active IO threads" logic
2. E2E testing the whole IO thread flow with a DEBUG command that sends commands while adjusting IO thread count up and down

But we can follow up on those improvements

WDYT?

Reply:
That is a good point. I think it would be good to address it separately though. Let's open a separate issue?

For this change, I am mostly interested in proving that the EMA logic is working properly (it can scale up and also scale back down) so I think a real unit test would be a lot less flaky to maintain in the long run.

Reply:
@PingXie https://github.com/valkey-io/valkey/issues/396 https://github.com/valkey-io/valkey/issues/397 were created to track the debug commands and tests. @murphyjacob4 please feel free to add. BTW, I have no permission to assign the issues.